### DISCUSSION TITLE
Using softlinks for pointers to libraries

### DISCUSSION THREAD

**Discussion by simon-streamotion** [elyra-ai/elyra#1646] [Q&A] [answered]
I'd like to have a directory setup something similar to the following:
```
root
  -- common_libs
  -- pipe_1
     -- pipe1_files
     -- softlink_to_../common_libs
  -- pipe_2 
     -- pipe2_files
     -- softlink_to_../common_libs
```
The idea is that there is just one common_libs directory containing all my common libraries, and each pipeline dir contains a soflink to that directory. That way, I can easily import the common libs into the stages of my pipeline.

The problem arises when I try to run the pipeline on kubeflow - the stages fail when trying to import anything from common_libs because it cannot find the file. 

It appears the tar command used to create the archive for the pipeline stages includes the softlink to common_libs rather than the files themselves. This is the default behaviour of tar but I believe it can be changed by using the -`-dereference` flag - which includes a copy of the file in the archive rather than the link.

Any chance of changing the tar command for building the stage archives to include the dereference flag ?

Many thanks -- Simon

**Comment by lresende** [comment]
@simon-streamotion Thanks for the interesting scenario, Elyra can certainly be updated to support it. As usual, you can provide a pull request, or we will add to our milestones to take a look at it.

**Comment by lresende** [comment]
Also, if you are interested in providing a patch, this is most likely the location that needs to be updated:
https://github.com/elyra-ai/elyra/blob/master/elyra/util/archive.py

**Reply by lresende** [reply]
BTW, I didn't realize this was just a discussion, let me transfer this to an issue.

**Reply by lresende** [reply]
It doesn't seem I can just convert this to an issue. @simon-streamotion would you mind creating a new issue, basically by copying the title and description to the new issue.

**Reply by simon-streamotion** [reply]
No worrries @lresende.
Created issue https://github.com/elyra-ai/elyra/issues/1648

**Comment by ptitzler** [comment, marked as answer]
Resolved in version 3.0+ by https://github.com/elyra-ai/elyra/pull/1689

**Reply by simon-streamotion** [reply]
@ptitzler & team - many thanks for fixing this :-)
